### PR TITLE
Wishlist

### PR DESCRIPTION
**Pull request by partha8**
## files to check

- [src/components/Card/Card.js](https://github.com/partha8/algern-clothing-react/compare/wishlist?expand=1#diff-c140b1cd54cc14c0d77d8af6d1b69d4a1e5456147d4379c8fcb3629e651d0392)
- [src/hooks/useGetWishlist.js](https://github.com/partha8/algern-clothing-react/compare/wishlist?expand=1#diff-e5fbfefdb119ea228b3576175558ebb85141918c413f6781e002cedaa22670ac) this will fetch the wishlist every time a user logs in and sets ```[]``` when no user is logged in.
- [src/pages/Wishlist/Wishlist.js](https://github.com/partha8/algern-clothing-react/compare/wishlist?expand=1#diff-c376116622abb6e8e9da10a0a62f01640f8fa213dc9f1ac4dc0346e5ff7e8923)
- [src/reducers/productsReducer.js](https://github.com/partha8/algern-clothing-react/compare/wishlist?expand=1#diff-1e72764740711c06e4f3cf410674a7d5a3bb67e5236ee138c5019d857f828e3a)
- [src/routes/PrivateRoute.js](https://github.com/partha8/algern-clothing-react/compare/wishlist?expand=1#diff-53a73b18f13b9ebe7433166a2c4ac09a844024a4f807ee3efe5acf1a675dba80) 
- [src/utils/productUtils.js](https://github.com/partha8/algern-clothing-react/compare/wishlist?expand=1#diff-316a67f1af7813b6dd0bddc5262ea859f542723384a44aeb1e8daed288ab6aa7)

## files to avoid
- [src/hooks/useSignup.js](https://github.com/partha8/algern-clothing-react/compare/wishlist?expand=1#diff-8225b8e14d75f7579754c1eb882205ebd88b28ce8246d009d185053ae0b45971)

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,21 +1,30 @@
...
 import { Login, SignUp } from "./pages/AuthPages/";
 
 import { useStateContext } from "./context/StateProvider";
 
-import { useGetCategories, useGetProducts } from "./hooks/";
+import {
+  useGetCategories,
+  useGetProducts,
+  useGetWishlist,
+  // useSignup,
```
Comment (nayyyhaa) [inline src/App.js:15]: it's good that you've used custom hooks here.

File: src/hooks/useGetWishlist.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    if (userState._id) {
+      (async () => {
+        try {
+          const response = await axios.get("/api/user/wishlist", {
+            headers: {
+              authorization: encodedToken,
+            },
+          });
+          console.log(response.status);
+          console.log(response.data);
```
Comment (nayyyhaa) [inline src/hooks/useGetWishlist.js]: can remove console.logs() later (if not required anymore)

**Review by nayyyhaa — APPROVED** [review]
Not much to comment on. Well-written code with good usage of custom hooks + proper logic! 👍

File: src/hooks/useGetWishlist.js (hunk 2 above)
Comment (partha8) [reply src/hooks/useGetWishlist.js]: yup, missed them. Thank you!

**Review by yash9263 — APPROVED** [review]

**Comment by yash9263** [comment]
I found nothing to suggest 😅 everything is well written, crisp, and clear.
Nice work

**Pull request merged**